packages/recommend/src/types/RecommendationsQuery.ts

```diff
@@ -15,7 +15,7 @@ export type RecommendationsQuery = {
   /**
    * The `objectID` of the item to get recommendations for.
    */
-  readonly objectID: string;
+  readonly objectID?: string;
```

Comment:
it should be required, but omitted from the new items then probably, as how facetName is omitted for fbt

Reply:
It is omitted actually, how come it's still needed?

Reply:
the problem if I do that is with getRecommendations . getRecommendation is used for every get method of Recommend and the objectID will be mandatory event for trends

Reply:
Can we still remodel the code so that types are accurate?

Reply:
I think so yes, if it does not break the current implementation. I'll do that. Thank you!

Reply:
So after a third though, I wanted to try to make it optional here but required for RP and FBT using `required`. I did that to be able to use `getRecommendations` for all models.

Reply:
Put differently: was `getRecommendation` a hack while we defined the actual model names before GA, or is it a feature designed to allow using not-yet-public new models?
- If the former, then it might be time we drop this generic method (which brings subpar "common denominator" DX compared to specialized methods) and only implement Trending as a new, fully-typed method
- If the latter, we'll need to find a generic implementation that is least disruptive to current users.

Reply:
`getRecommendations()` is useful for the situation we're in right now: users can already use the trending model, without it being officially supported with a method in the client.

We didn't have a team to maintain this Recommend client initially so we provided it as an "escape hatch" if we're not reactive to support new models in the client.

`getRecommendations()` was just the very initial implementation that satisfied us at the time, feel free to change it as you need. And if it's not helpful for the new model, don't feel like you need to use it. (In the future, we can also deprecate it.)

Reply:
I'll ask the opposite question then: what's the purpose of having a specialized method for each model rather than just using `getRecommendations()`? It's basically a multi-query endpoint, and I think it has its uses (like making a single call to retrieve both trending items and trending facets for example). While calling `getRelatedProducts()` as a multi-query is strange IMO.